doc/lsst.pipe.base/creating-a-pipeline.rst

```diff
@@ -0,0 +1,448 @@
+simple enough that it does not warrant its own config file. The ``python``
+key is designed to support this use case. The value associated with the key
+is a (possibly multi-line) string with valid python syntax. This string is
+evaluated and applied during task configuration exactly as if it had been
+written in a file or typed out in an interpreter. The following example expands
+the previous one to use the ``python`` key.
+
+.. code-block:: yaml
+
+  makeWarp:
+    class: lsst.pipe.tasks.makeCoaddTempExp.MakeWarpTask
+    config:
+      matchingKernelSize: 29
+      makePsfMatched: true
+      modelPsf.defaultFwhm: 7.7
+      doApplyExternalPhotoCalib: false
+      doApplyExternalSkyWcs: false
+      doApplySkyCorr: false
+      doWriteEmptyWarps: true
+      python: "config.warpAndPsfMatch.psfMatch.kernel['AL'].alardSigGauss = \
+        [1.0, 2.0, 4.5]"
+
+----------
+Parameters
+----------
+
+As you saw in the pervious section, each task defined in a `Pipeline` may
+have its own configuration. However, it is sometimes useful for configuration
+fields in multiple tasks to share the same value. `Pipeline`\ s support this
+with a concept called ``parameters``. This is a top level section in the
+`Pipeline` document specified with a key named ``parameters``.
+
+The contents of the ``parameters`` section is a mapping of key, value pair-
+associated value in the parameters section.
+
+To make this a bit clearer take a look at the following example, making note
+that only config fields relevant for this example are shown.
+
+.. code-block:: yaml
+
+  parameters:
+    calibratedSingleFrame: calexp
+  tasks:
+    calibrate:
+      class: lsst.pipe.tasks.calibrate.CalibrateTask
+      config:
+        connections.outputExposure = parameters.calibratedSingleFrame
+    makeWarp:
+      class: lsst.pipe.tasks.makeCoaddTempExp.MakeWarpTask
+      config:
+        connections.calExpList = parameters.calibratedSingleFrame
+    forcedPhotCcd:
+      class: lsst.meas.base.forcedPhotCcd.ForcedPhotCcdTask
+      config:
+        connections.exposure = parameters.calibratedSingleFrame
+
+The above example used ``parameters`` to link the dataset type names for
+multiple tasks, but ``parameters`` can be used anywhere that more than one
+config field use the same value.
+
+:ref:`pipeline-running-intro` introduces how to run `Pipeline`\ s and will
+talk about how to dynamically set a parameters value at `Pipeline` invocation
+time.
+
+----------------------------------
+Verifying Configuration: Contracts
+----------------------------------
+
+The `~lsst.pipe.base.config.Config` classes associated with
+`~lsst.pipe.base.task.Task`\ s provide a method named ``verify`` which can be
+used to verify that all supplied configuration is valid. These verify methods
+however, are shared by every instance of the config class. This means they
+can not be specialized for the context in which the task is being used.
```

Comment:
I put this in specifically because I was asked to mention that this is separate from verify in configs. I think that it sets up a bit of understanding as well that contracts are useful for multiple task (and later cross task) verification. I will look to see if there is any other way to work this in though. However I am assuming some knowledge of some LSST software from the readers.